spec/models/manageiq/providers/vmware/infra_manager/provision_workflow_spec.rb

```diff
@@ -93,83 +93,35 @@
+        @src_vm.host.switches = [s11, s12, s13, s14]
+        @lan11 = FactoryGirl.create(:lan, :name => "lan_A",   :switch_id => s11.id)
+        @lan12 = FactoryGirl.create(:lan, :name => "lan_B",   :switch_id => s12.id)
+        @lan13 = FactoryGirl.create(:lan, :name => "lan_C",   :switch_id => s13.id)
+        @lan14 = FactoryGirl.create(:lan, :name => "lan_DVS", :switch_id => s14.id)
+        allow(workflow).to receive(:allowed_hosts).with(no_args).and_return([workflow.host_to_hash_struct(@host1)])
       end
 
       it '#allowed_vlans' do
-        allow(workflow).to receive(:allowed_hosts).with(no_args).and_return([workflow.host_to_hash_struct(@host1)])
+        vlans, _hosts = workflow.allowed_vlans(:vlans => true, :dvs => true)
+        lan_keys   = [@lan11.name, @lan13.name, @lan12.name, "dvs_#{@lan14.name}"]
+        lan_values = [@lan11.name, @lan13.name, @lan12.name, "#{@lan14.name} (#{s14.name})"]
+        expect(vlans.keys).to match_array(lan_keys)
+        expect(vlans.values).to match_array(lan_values)
+      end
+
+      it 'excludes dvs if told so' do
         vlans, _hosts = workflow.allowed_vlans(:vlans => true, :dvs => false)
         lan_keys = [@lan11.name, @lan13.name, @lan12.name]
         expect(vlans.keys).to match_array(lan_keys)
         expect(vlans.values).to match_array(lan_keys)
       end
-    end
 
-    context 'dvs' do
-      before do
-        @host1_dvs = {'pg1' => ['switch1'], 'pg2' => ['switch2']}
-        @host1_dvs_hash = {'dvs_pg1' => 'pg1 (switch1)',
-                           'dvs_pg2' => 'pg2 (switch2)'}
-        allow_any_instance_of(ManageIQ::Providers::Vmware::InfraManager).to receive(:connect)
-        allow(workflow).to receive(:get_host_dvs).with(@host1, nil).and_return(@host1_dvs)
-      end
-
-      it '#allowed_dvs single host' do
-        workflow.instance_variable_set(:@values, :vm_tags => [], :src_vm_id => @src_vm.id,
-                                      :host_id => @host1.id)
-        workflow.instance_variable_set(:@target_resource,
-                                       :host    => workflow.host_to_hash_struct(@host1),
-                                       :ems     => workflow.ci_to_hash_struct(@ems),
-                                       :host_id => @host1.id)
-        dvs = workflow.allowed_dvs({}, nil)
-        expect(dvs).to eql(@host1_dvs_hash)
-      end
-
-      context "#allowed_dvs" do
-        before do
-          @host2 = FactoryGirl.create(:host_vmware, :ems_id => @ems.id)
-          @host2_dvs = {'pg1' => ['switch21'], 'pg2' => ['switch2'], 'pg3' => ['switch23']}
-          allow(workflow).to receive(:get_host_dvs).with(@host2, nil).and_return(@host2_dvs)
-          workflow.instance_variable_set(:@values, :vm_tags => [], :src_vm_id => @src_vm.id,
-                                        :placement_auto => true)
-
-          @combined_dvs_hash = {'dvs_pg1' => 'pg1 (switch1/switch21)',
```

Comment:
@jameswnl @gmcculloug If we're just keying off of the name of the portgroup, what happens when you have two different portgroups (different ems_refs) on different switches with the same name?  It looks like this will combine the two portgroups into one on multiple dvswitches.